lib/block-supports/layout.php

```diff
@@ -86,7 +86,14 @@ function gutenberg_render_layout_support_flag( $block_content, $block ) {
 		1
 	);
 
-	return $content . '<style>' . $style . '</style>';
+	add_action(
+		'wp_head',
+		function () use ( $style ) {
+			echo '<style>' . $style . '</style>';
+		}
+	);
```

Comment:
I do believe using `wp_head` here will only work for block based themes.

On classic themes the content isn't parsed after `wp_head` is already runned so this will never get printed. Basically the same reason global styles are loaded in the footer when loading core assets individually. See https://core.trac.wordpress.org/changeset/51309